compiler/rustc_mir_transform/src/jump_threading.rs

```diff
@@ -388,24 +392,30 @@ impl<'a, 'tcx> TOFinder<'a, 'tcx> {
             lhs,
             constant,
             &mut |elem, op| match elem {
-                TrackElem::Field(idx) => self.ecx.project_field(op, idx.as_usize()).ok(),
-                TrackElem::Variant(idx) => self.ecx.project_downcast(op, idx).ok(),
+                TrackElem::Field(idx) => {
+                    self.ecx.project_field(op, idx.as_usize()).discard_interp_error()
```

Comment:
Bikeshed: If we *can't* get rid of all these calls in rustc_mir_transform, perhaps we could at least find a shorter name than `discard_interp_error`.

Reply:
The name must be somewhat scary though or else people might use it in const-eval / miri and not realize what they are doing (and reviewers could easily miss it)...